lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -83,10 +90,10 @@ def user_can_access_course(self, user, course):
     def check_course_permissions(self, user, course):
         """
         Checks if the request user can access the course.
-        Raises PermissionDenied if the user does not have course access.
+        Raises 404 if the user does not have course access.
         """
         if not self.user_can_access_course(user, course):
-            raise PermissionDenied
+            raise Http404
```

Comment:
@clintonb Can you confirm that this change is acceptable for the existing Course-Structure API?  Elsewhere in the platform we raise a 404 when the user cannot access a course.  This is what we've recommended in edX API Conventions document.